xbmc/guilib/guiinfo/PlayerGUIInfo.cpp

```diff
@@ -66,7 +68,9 @@ float CPlayerGUIInfo::GetSeekPercent() const
 
   float fPercentPlayTime = static_cast<float>(GetPlayTime() * 1000) / iTotal * 0.1f;
   float fPercentPerSecond = 100.0f / static_cast<float>(iTotal);
-  float fPercent = fPercentPlayTime + fPercentPerSecond * g_application.GetAppPlayer().GetSeekHandler().GetSeekSize();
+  const auto& components = CServiceBroker::GetAppComponents();
+  const auto appPlayer = components.GetComponent<CApplicationPlayer>();
```

Comment:
Looking at so many of these lines in this source file, I would introduce a local helper function `GetAppPlayer`... Applies for all other files where now many of those two-liners appear.

Reply:
Without getting personal, but this whole construct of having to call these functions every time again seems to me to be more than brainless and EXTREMELY ineffective. It seems to me that everything has to be programmed as awkwardly as possible in order to waste as much CPU resources as possible. I admit, I'm not a C++ programmer and I never will be, I come from embedded programming, where it's always been about effectiveness. in the PR https://github.com/xbmc/xbmc/pull/21934 appPower will be added, and in the following PR's probably something new again. You'll go crazy with time.

I'm sorry, but I just had to get that off my chest.

Reply:
@mkreisl Your views here are appreciated, embedded brings a diversity that is more than welcome. One of the things I learned, coming from embedded, is that efficiency is one of lesser values when organizing a codebase of 500K lines with hundreds of contributors.

In our case, architecture is king and has been key to the long-term maintainability of the codebase. Moreso, as you pick up more C++, you'll see that most of this "bloat" you see compiles down to the same underlying code, and where it doesn't, unless it's a hot path, it really doesn't matter.

Seeing as how this PR is entirely about architecture, spearheaded by one of our leading architects, I suggest we value it on those principals instead of any perceived "ineffectiveness", unless numbers are provided to back up such perceptions.

Reply:
discussed on slack. will hold off on helpers for now.